File: tests/fixtures/edgecase_repo/cpp/module4.cpp
#include <string>
#include "module3.hpp"

namespace gen {

class Base4 {
public:
    std::string name;
};

class Service4 : public Base4 {
public:
    bool handle(const std::string &payload) {
        this->validate(payload);
        return true;
    }

    void validate(const std::string &payload) {}
};

int helper_4(int value) { return value; }

}  // namespace gen

File: tests/fixtures/edgecase_repo/ruby/module4.rb
require_relative "./module3"

module Example
  class Base4
    def initialize(name)
      @name = name
    end
  end

  class Service4 < Base4
    def handle(payload)
      validate(payload)
      true
    end

    def validate(payload)
      !payload.nil?
    end
  end
end

def helper_4(value)
  value
end
